Bluepill-cli/Bluepill-cli/Simulator/BPSimulator.m

```diff
@@ -79,29 +79,29 @@ - (NSString *)getErrorDescription:(NSError *__autoreleasing *)error {
     return error != nil ? [*error localizedDescription] : nil;
 }
 
-- (NSString *)installApplicationWithHost:(NSString *)testHost withError:(NSError *__autoreleasing *)error {
-    SimServiceContext *sc = [SimServiceContext sharedServiceContextForDeveloperDir:self.config.xcodePath error:error];
+- (NSString *)installApplicationWithHost:(NSString *)testHost withError:(NSError *__autoreleasing *)errPtr {
+    SimServiceContext *sc = [SimServiceContext sharedServiceContextForDeveloperDir:self.config.xcodePath error:errPtr];
     if (!sc) {
-        [BPUtils printInfo:ERROR withString:@"%@", [NSString stringWithFormat:@"SimServiceContext failed: %@", [self getErrorDescription:error]]];
+        [BPUtils printInfo:ERROR withString:@"%@", [NSString stringWithFormat:@"SimServiceContext failed: %@", [self getErrorDescription:errPtr]]];
         return nil;
     }
-    SimDeviceSet *deviceSet = [sc defaultDeviceSetWithError:error];
+    SimDeviceSet *deviceSet = [sc defaultDeviceSetWithError:errPtr];
     if (!deviceSet) {
-        [BPUtils printInfo:ERROR withString:@"%@", [NSString stringWithFormat:@"SimDeviceSet failed: %@", [self getErrorDescription:error]]];
+        [BPUtils printInfo:ERROR withString:@"%@", [NSString stringWithFormat:@"SimDeviceSet failed: %@", [self getErrorDescription:errPtr]]];
         return nil;
     }
     SimDevice *simDevice = [deviceSet createDeviceWithType:self.config.simDeviceType
                                                    runtime:self.config.simRuntime
                                                       name:@"BP_Simultator_Template"
-                                                     error:error];
+                                                     error:errPtr];
     [self.simDeviceTemplates addObject:simDevice];
-    if (!simDevice || error) {
-        [BPUtils printInfo:ERROR withString:@"Create simulator failed with error: %@", [self getErrorDescription:error]];
+    if (!simDevice || errPtr) {
```

Comment:
Renaming of this value showcases yet another issue ( my PR missed this ). Here, it seems like this should *check the pointee* of the errPtr it’s not nil.

Reply:
`getErrorDescription` handles the case if `errPtr` is `nil`. 

I guess you mean what if `simDevice` is nil but errPtr is not nil. This is unlikely to happen unless `SimDevice` is not properly written. Also, it seems we don't check if error is nil before printing its debugging information throughout the code base, also I don't know what else we could spit out if both simDevice & error are nil

Reply:
I was thinking of the case when errPtr isn’t nil: when a caller passes a non-nil errPtr pointer that points to nil.

It seems like some of the logic in this method would be more concerned with checking the value of the pointer ( e.g. by dereferencing it to see if an error has happened and is set ).

Otherwise, I’m terribly confused 🤯🥳

Reply:
oh.. i see what you mean now.. that makes sense.

Reply:
This turned out to be a tricky issue to fix if we want to do it the ideal way.

Ideally: 
1) Add `nonnull` decorator to make sure user doesn't pass in a nil
2) For every invocation taking an errorPtr, we create an `NSError *error` first and log it. If the errPtr is not null, assign *errPtr to that error..

Not planning to do either in this PR. Gonna assume user would always pass in a non 0 errorPtr for now 🤔